### DISCUSSION TITLE
Ubuntu 22.04 Guide

### DISCUSSION THREAD

**Discussion by MaxflowO2** [ordinals/ord#1696] [Q&A]
Download bicoin core from *[here](https://github.com/bitcoin/bitcoin)*
Then preform `git checkout 2b87e905857ba456db2cb3d0297a3102f17417be` for latest 24.x
Follow the guide found *[here](https://baloian.medium.com/how-to-setup-and-run-a-bitcoin-full-node-on-ubuntu-a106fb86dbb3)*
Install bitcoind.service [advanced if you wish](https://bitcoin.stackexchange.com/questions/13795/ubuntu-linux-how-do-i-start-bitcoind-as-a-service-to-run-automatically)
Download ord from *[here](https://github.com/casey/ord)*
Then preform:
```
sudo apt-get install libssl-dev
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```
You should get this prompt now...
```
To get started you may need to restart your current shell.
This would reload your PATH environment variable to include
Cargo's bin directory ($HOME/.cargo/bin).

To configure your current shell, run:
source "$HOME/.cargo/env"
```
Run that last line then the following
```
git clone https://github.com/casey/ord.git
cd ord
cargo build --release
```
When you type ord, you will get errors like this below
```
error: DB corrupted: Expected file format version 107, found 109
```
Simply wipe index.redb
```
cd ~/.local/share/ord/
rm index.redb
```
Now run this
```
ord index
[indexing blocks] ████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ 51768/776256
```
Later run...
```
ord --rpc-url localhost:8332 server --address localhost --http-port (port)
Listening on http://127.0.0.1:(port)
```
Use a proxy pass through and you get a nice block explorer...

**Comment by MaxflowO2** [comment]
Side note:
[This applies](https://github.com/casey/ord/discussions/1619) as well as [This](https://github.com/casey/ord/discussions/1739)

**Comment by AdrianScott** [comment]
Excellent; any hardware configuration recommendations/minimums? Especially for slow ord startup (when already indexed)

**Reply by MaxflowO2** [reply]
https://github.com/casey/ord/discussions/1791 explains it all
